AmplifyPlugins/DataStore/Sources/AWSDataStorePlugin/Storage/CascadeDeleteOperation.swift

```diff
@@ -144,12 +148,25 @@ public class CascadeDeleteOperation<M: Model>: AsynchronousOperation {
         
         associatedModels = await self.recurseQueryAssociatedModels(modelSchema: self.modelSchema, ids: modelIds)
         
+        if mutationEvent != "MutationEvent" {
```

Comment:
w.r.t cascadeDelete Operation, logging the actual models(like Post, Blog etc) and skipping MutationEvent(here a form of model) is both cx and developer friendly to only track the state of cascadeDelete operation of the required actual model. 
If I don't skip logging on MutationEvents, my logs look like:
```
[CascadeDelete.1] Start cascade logic for MutationEvent with id ["456"]
[CascadeDeleteOperation<MutationEvent>] [CascadeDelete.2] Query for MutationEvent associated models, found []
[CascadeDeleteOperation<MutationEvent>] [CascadeDelete.3] Local cascade delete of MutationEvent successful!
[CascadeDeleteOperation<MutationEvent>] [CascadeDelete.1] Start cascade logic for MutationEvent with id ["567"]
[CascadeDeleteOperation<MutationEvent>] [CascadeDelete.2] Query for MutationEvent associated models, found []
[CascadeDeleteOperation<MutationEvent>] [CascadeDelete.3] Local cascade delete of MutationEvent successful!
[CascadeDeleteOperation<MutationEvent>] [CascadeDelete.1] Start cascade logic for MutationEvent with id ["678"]
[CascadeDeleteOperation<MutationEvent>] [CascadeDelete.2] Query for MutationEvent associated models, found []
[CascadeDeleteOperation<MutationEvent>] [CascadeDelete.3] Local cascade delete of MutationEvent successful!
[CascadeDeleteOperation<Blog6V2>] [CascadeDelete.1] Start cascade logic for Blog with id ["123"]
[CascadeDeleteOperation<Blog6V2>] [CascadeDelete.2] Query for Blog associated models, found ["Post", "Comment"]
[CascadeDeleteOperation<Blog6V2>] [CascadeDelete.3] Local cascade delete of Blog6V2 successful!
[CascadeDeleteOperation<Blog6V2>] [CascadeDelete.4] sending 3 delete mutations
```
The filtering of MutationEvents(here a form of model) is to refine the logs of required models & show their state. for example, with filtering the MutationEvents, the logs look clean as in:
```
[CascadeDeleteOperation<Blog6V2>] [CascadeDelete.1] Start cascade logic for Blog with id ["123"]
[CascadeDeleteOperation<Blog6V2>] [CascadeDelete.2] Query for Blog associated models, found ["Post", "Comment"]
[CascadeDeleteOperation<Blog6V2>] [CascadeDelete.3] Local cascade delete of Blog6V2 successful!
[CascadeDeleteOperation<Blog6V2>] [CascadeDelete.4] sending 3 delete mutations
```

Reply:
filtering out the MutationEvent deletions seems fair, let's do this for the remaining system model scehams then. Let's create a computed property called `isDeveloperDefinedModel` in this class that checks for the model names of the system models

```
ModelSyncMetadata.modelName
MutationEvent.modelName
MutationSyncMetadata.modelName
```
 so that we can reuse wherever you want to create the developer friendly logging. We also should not be hardcoding the string "MutationEvent" and use the model's properties directly. Hardcoding a string multiple times is prone to typos and if the name changes, this string needs to change as well, but is not enforced by the compiler.